src/custom/pages/NewSwap/hooks/useSetupSwapState.test.tsx

```diff
@@ -0,0 +1,251 @@
+import { unmountComponentAtNode } from 'react-dom'
+import { useSetupSwapState } from 'pages/NewSwap/hooks/useSetupSwapState'
+import { useSwapState } from 'state/swap/hooks'
+import { waitFor } from '@testing-library/react'
+import { mockedConnector, mockedWeb3Render } from 'test-utils'
+import { createHashHistory } from 'history'
+import { Field, replaceSwapState, ReplaceSwapStatePayload } from '@src/state/swap/actions'
+import store from 'state'
+
+function TestComponent() {
+  useSetupSwapState()
+  const swapState = useSwapState()
+
+  return <div id="result">{JSON.stringify(swapState)}</div>
```

Comment:
Maybe this was done because you needed to wrap the component to mock the web3 connection? 

I have not too much experience on this, but could this be a solution? https://react-hooks-testing-library.com/usage/advanced-hooks#context

Reply:
🤦‍♂️ I invented the wheel
Thank you! Used this tools